src/lib/libcore.js

```diff
@@ -2296,6 +2296,62 @@ addToLibrary({
 
   $noExitRuntime: "{{{ makeModuleReceiveExpr('noExitRuntime', !EXIT_RUNTIME) }}}",
 
+  // The following addOn<X> functions are for adding runtime callbacks at
+  // various executions points. Each addOn<X> function has a corresponding
+  // compiled time version named addAt<X> that will instead inline during
+  // compilation (see parseTools.mjs).
+  // Note: if there are both runtime and compile time code, the runtime
+  // callbacks will be invoked before the compile time code.
+
+  // See ATPRERUNS in parseTools.mjs for more information.
+  $__ATPRERUN__: [],
+  $__ATPRERUN____deps: ['$callRuntimeCallbacks'],
+  $__ATPRERUN____postset: () => {
+    ATPRERUNS.unshift('callRuntimeCallbacks(__ATPRERUN__);');
+  },
+  $addOnPreRun__deps: ['$__ATPRERUN__'],
+  $addOnPreRun: (cb) => __ATPRERUN__.unshift(cb),
+  // See ATINITS in parseTools.mjs for more information.
+  $__ATINIT__: [],
+  $__ATINIT____deps: ['$callRuntimeCallbacks'],
+  $__ATINIT____postset: () => {
+    ATINITS.unshift('callRuntimeCallbacks(__ATINIT__);');
+  },
+  $addOnInit__deps: ['$__ATINIT__'],
+  $addOnInit: (cb) => __ATINIT__.unshift(cb),
+  // See ATPOSTCTORS in parseTools.mjs for more information.
+  $__ATPOSTCTOR__: [],
+  $__ATPOSTCTOR____deps: ['$callRuntimeCallbacks'],
+  $__ATPOSTCTOR____postset: () =>  {
+    ATPOSTCTORS.unshift('callRuntimeCallbacks(__ATPOSTCTOR__);');
+  },
+  $addOnPostCtor__deps: ['$__ATPOSTCTOR__'],
+  $addOnPostCtor: (cb) => __ATPOSTCTOR__.unshift(cb),
+  // See ATMAINS in parseTools.mjs for more information.
+  $__ATMAIN__: [],
+  $__ATMAIN____deps: ['$callRuntimeCallbacks'],
+  $__ATMAIN____postset: () => {
+    ATMAINS.unshift('callRuntimeCallbacks(__ATMAIN__);');
+  },
```

Comment:
These can be one-liners too I think?

Reply:
I tried, but it doesn't work because the return value of `unshift` is the length of the array, so that gets put into the js file as a postset. At least the one liner here doesn't matter for output code size.

Reply:
Ah yes, that old double meaning of `__postset`..   grrr.